### DISCUSSION TITLE
Voiding an invoice

### DISCUSSION THREAD

**Discussion by ajakwani** [netsuitephp/netsuite-php#173] [General]
How do I void an invoice? I can do it in NetSuite so there are no issues with turning on any features in NetSuite. What have I tried so far is changing the status of transaction and putting the status of invoiceVoided but it does not change anything. It is not returning any errors but the status or the transaction state is not changing in NetSuite.

`        $service = app(NetSuiteService::class);
        $netSuiteInvoice = new Invoice();
        $netSuiteInvoice->internalId = 127667;
        $netSuiteInvoice->status = TransactionStatus::_invoiceVoided;

        $request = new UpdateRequest();
        $request->record = $netSuiteInvoice;

        $result = $service->update($request);`

**Comment by jrebs** [comment]
At a glance I think the request looks OK. It might help if you included the full response body.

**Reply by ajakwani** [reply]
Well, I inspected the response as well. It was returning the exact same status as before with a success status of the request. I was confused that is it even the correct way to void a transaction.

**Reply by jrebs** [reply]
> Well, I inspected the response as well. It was returning the exact same status as before with a success status of the request. I was confused that is it even the correct way to void a transaction.

I can only guess since I've never voided transactions (I've just deleted them when anything like that was needed, and deleting has worked fine). Maybe someone else will have some suggestion. Generally if you're getting back a success response, your request is OK. If a transaction is closed for modification (because the accounting period has been closed off), you should get back an API error, so if you're getting OK but no changes...not sure. Good luck.

**Reply by ajakwani** [reply]
Okay Thank You.
